File: app/src/main/java/com/example/solar_alarm/Data/AlarmRepository.java
package com.example.solar_alarm.Data;

import android.app.Application;
import androidx.lifecycle.LiveData;
import java.util.List;

public class AlarmRepository {
    private AlarmDao alarmDao;
    private LiveData<List<Alarm>> alarmsLiveData;

    public AlarmRepository(Application application) {
        AlarmDatabase db = AlarmDatabase.getDatabase(application);
        alarmDao = db.alarmDao();
        alarmsLiveData = alarmDao.getAlarms();
    }

    public void insert(Alarm alarm) {
        AlarmDatabase.databaseWriteExecutor.execute(() -> {
            alarmDao.insert(alarm);
        });
    }

    public void update(Alarm alarm) {
        AlarmDatabase.databaseWriteExecutor.execute(() -> {
            alarmDao.update(alarm);
        });
    }

    public void delete(Alarm alarm)
    {
        AlarmDatabase.databaseWriteExecutor.execute(() -> {
            alarmDao.delete(alarm);
        });
    }

    public void deleteAll()
    {
        AlarmDatabase.databaseWriteExecutor.execute(() -> {
            alarmDao.deleteAll();
        });
    }

    public LiveData<List<Alarm>> getAlarmsLiveData() {
        return alarmsLiveData;
    }
}

File: app/src/main/java/com/example/solar_alarm/Data/AlarmRepository.kt
package com.example.solar_alarm.Data

import androidx.annotation.RequiresApi
import android.os.Build
import android.app.Application
import androidx.lifecycle.LiveData

@RequiresApi(Build.VERSION_CODES.O)
class AlarmRepository(application: Application) {
    private val alarmDao: AlarmDao?
    val alarmsLiveData: LiveData<List<Alarm?>?>?

    init {
        val db: AlarmDatabase? = AlarmDatabase.Companion.getDatabase(application)
        alarmDao = db?.alarmDao()
        alarmsLiveData = alarmDao?.alarms
    }

    fun insert(alarm: Alarm?) {
        AlarmDatabase.Companion.databaseWriteExecutor.execute(Runnable { alarmDao!!.insert(alarm) })
    }

    fun update(alarm: Alarm?) {
        AlarmDatabase.Companion.databaseWriteExecutor.execute(Runnable { alarmDao!!.update(alarm) })
    }

    fun delete(alarm: Alarm?) {
        AlarmDatabase.Companion.databaseWriteExecutor.execute(Runnable { alarmDao!!.delete(alarm) })
    }

    fun deleteAll() {
        AlarmDatabase.Companion.databaseWriteExecutor.execute(Runnable { alarmDao!!.deleteAll() })
    }
}